Data/models/CRUDProtocols.swift

```diff
@@ -27,8 +27,8 @@ public protocol Readable where Self: NSManagedObject {
 
 // MARK: - Implementations
 public extension Deletable where Self: NSManagedObject {
-    func delete() {
-        let context = DataController.newBackgroundContext()
+    func delete(context: NSManagedObjectContext? = nil) {
```

Comment:
Why not just make the default `DataController.newBackgroundContext()`? Then don't need the line below, and can remove optionality.